table/mock_table.cc

```diff
@@ -27,6 +27,154 @@ stl_wrappers::KVMap MakeMockFile(
   return stl_wrappers::KVMap(l, stl_wrappers::LessOfComparator(&icmp_));
 }
 
+class MockTableReader : public TableReader {
```

Comment:
Warning:

Encapsulate structs and classes defined in cpp files with anonymous namespace
Struct and class definitions should either be defined in a header file, declared in a header file with the same base name and path as the definition file, or in an anonymous namespace to prevent possible shadowing of other classes.

Reply:
There are similar calls in other new classes in the file.